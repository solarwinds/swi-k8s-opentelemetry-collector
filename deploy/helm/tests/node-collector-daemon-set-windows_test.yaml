# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Test for node-collector-daemon-set-windows
templates:
  - node-collector-daemon-set-windows.yaml
  - node-collector-config-map-windows.yaml
  - common-env-config-map.yaml
tests:
  - it: DaemonSet spec for windows nodes should match snapshot when using default values
    template: node-collector-daemon-set-windows.yaml
    asserts:
      - matchSnapshot:
          path: spec.template.spec