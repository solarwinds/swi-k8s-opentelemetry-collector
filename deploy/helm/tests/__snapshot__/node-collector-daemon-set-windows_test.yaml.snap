DaemonSet spec for windows nodes should match snapshot when using default values:
  1: |
    affinity:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
            - matchExpressions:
                - key: eks.amazonaws.com/compute-type
                  operator: NotIn
                  values:
                    - fargate
    containers:
      - command:
          - c:\wrapper.exe
          - c:\swi-otelcol.exe
          - --config=c:\conf\relay.yaml
        env:
          - name: CHECKPOINT_DIR
            value: c:/var/lib/swo/checkpoints
          - name: POD_NAME
            valueFrom:
              fieldRef:
                fieldPath: metadata.name
          - name: POD_NAMESPACE
            valueFrom:
              fieldRef:
                fieldPath: metadata.namespace
          - name: NODE_NAME
            valueFrom:
              fieldRef:
                fieldPath: spec.nodeName
          - name: SOLARWINDS_API_TOKEN
            valueFrom:
              secretKeyRef:
                key: SOLARWINDS_API_TOKEN
                name: solarwinds-api-token
                optional: true
        envFrom:
          - configMapRef:
              name: RELEASE-NAME-swo-k8s-collector-common-env
        image: solarwinds/swi-opentelemetry-collector:0.8.8-nanoserver-ltsc2022
        imagePullPolicy: IfNotPresent
        livenessProbe:
          httpGet:
            path: /
            port: 13133
        name: swi-opentelemetry-collector
        ports:
          - containerPort: 8888
            name: http
            protocol: TCP
        readinessProbe:
          httpGet:
            path: /
            port: 13133
        resources:
          limits:
            memory: 700Mi
          requests:
            memory: 50Mi
        volumeMounts:
          - mountPath: c:\var\log\pods
            name: varlogpods
            readOnly: true
          - mountPath: c:\var\log\containers
            name: varlogcontainers
            readOnly: true
          - mountPath: c:\var\lib\docker\containers
            name: varlibdockercontainers
            readOnly: true
          - mountPath: c:\conf
            name: opentelemetry-collector-configmap
            readOnly: true
          - mountPath: c:/var/lib/swo/checkpoints
            name: logcheckpoints
    nodeSelector:
      kubernetes.io/arch: amd64
      kubernetes.io/os: windows
    terminationGracePeriodSeconds: 30
    tolerations:
      - effect: NoSchedule
        operator: Exists
    volumes:
      - hostPath:
          path: c:\var\log\pods
        name: varlogpods
      - hostPath:
          path: c:\var\log\containers
        name: varlogcontainers
      - hostPath:
          path: c:\var\lib\docker\containers
        name: varlibdockercontainers
      - hostPath:
          path: c:/var/lib/swo/checkpoints/4d51349fb86079745afe70b8333fd96ee2255b88b63eb2b37417c92aefa73fcb
          type: DirectoryOrCreate
        name: logcheckpoints
      - configMap:
          items:
            - key: logs.config
              path: relay.yaml
          name: RELEASE-NAME-swo-k8s-collector-node-collector-config-windows
        name: opentelemetry-collector-configmap
